api/spec/api_docs/definitions/resource.rb

```diff
@@ -1,78 +1,149 @@
 module ApiDocs
-  module Definition
+  module Definitions
     module Resource
-      def resource_response
-        definition = Type.object(
-          properties: {
-            data: Type.object(properties: {
-                                id: Type.id,
-                                type: Type.string(example: type),
-                                attributes: Type.object(properties: response_attributes),
-                                relationships: response_relationships,
-                                meta: Type.meta
-                              })
-          }
+      ####################################
```

Comment:
Part of why this class is now so much more confusing is that you've greatly expanded its public interfaces. When I wrote this class, it had four interfaces: resource_response, collection_response, update_request, and create_request. It now has over 15 additional public interfaces. Why? Are the necessarily public, or did you just not consider whether they were public or private? Can you make it clear which of these methods are internal implementation details, and which are part of its public interface?

Reply:
That's a good point. I will be more cognizant of what methods are public and private in a class. I was trying to organize the methods so I could track the distinction between request and response functionality, but in doing so, I messed up the private and public statuses.